Add Checklist to PRs (#1041)
privacytoolsIO/privacytools.io: .github/PULL_REQUEST_TEMPLATE.md

## Description

Resolves: #none <!-- The number of the issue that is resolved by this pull request. If there is none, feel free to delete this line 

#### Check List <!-- Please add an x in each box below, like so: [x] -->

- [ ] I have read and understand [CONTRIBUTING.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md)

- [ ] I have listed the source code for this project in [source_code.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md)

- [ ] This project is [free/libre software](https://www.wikipedia.org/wiki/Free_software)

> Code Repository (if applicable): 

Reviewer: Since we have this open PR already, could we add

```markdown
[ ] This project has an [associated discussion](https://github.com/privacytoolsIO/privacytools.io/issues)
```

<!--
## Screenshots

> -->

Reviewer 2: This entire section (L7-L10) should just be removed entirely.

#### Check List <!-- Please add an x in each box below, like so: [x] -->

> - [ ] I Have read and understand [CONTRIBUTING.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md)

Reviewer 2: ```suggestion
- [ ] I have read and understand [CONTRIBUTING.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md)
```

> - [ ] I have listed any applicable source [source_code.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md)

Reviewer 2: ```suggestion
- [ ] I have listed the source code for this project in [source_code.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md)
```

> - [ ] This project is [Free Software](https://www.wikipedia.org/wiki/Free_software)

Reviewer 2: ```suggestion
- [ ] This project is [free software](https://www.wikipedia.org/wiki/Free_software)
```

Author: Thanks

> Please add screenshots if applicable

Reviewer 3: I wonder if this could be removed as Netlify handles that?

> Please add screenshots if applicable

Reviewer: ```suggestion
```

-->

> #### Check List

Reviewer 2: I would add instructions for those unfamiliar with Markdown.

```suggestion
#### Check List <!-- Please add an x in each box below, like so: [x] -->
```

> - [ ] Have read and understand [CONTRIBUTING.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md)

Reviewer 3: I think this is reduntant as when you open a PR, GitHub suggests you on the right to read the contributing guidelines. It also appears somewhat formal to me.

Reviewer 2: I think this is fine to include if we're doing this, because it means people need to affirm that they have read it directly.

```suggestion
- [ ] I have read and understand [CONTRIBUTING.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md)
```

Author: fixed in latest commit.

Thx

> - [ ] Listed Source code in [Source_Code.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md)

Reviewer 3: I am not sure if this is that good idea as it could easily cause a merge conflict if multiple users were submitting PRs to add multiple apps/services simultaneously. I think it should possibly be something for a team member to update (but it needs a PR with approval)? Ideas @BurungHantu1605 ?

Reviewer 2: AFAIK it will only cause a merge conflict if people edit the same line, which seems unlikely since they'll likely be editing different sections of the file. But in any case merge conflicts are relatively easy to solve when a team member merges them. I would however make the following grammatical changes:

```suggestion
- [ ] I have listed any applicable source code in [source_code.md](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md)
```

> - [ ] Project Is [Free Software](https://www.wikipedia.org/wiki/Free_software)

Reviewer 3: I think this should be "Project is preferably..."

Reviewer 2: ```suggestion
- [ ] This project is [free software](https://www.wikipedia.org/wiki/Free_software), unless 1) no free software alternative exists, or 2) this software is significantly better than a free software alternative.
```

Author: Software must actually be free software:
https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md#software

But, ya that description makes more sense.

> Code Repository (if applicable): 

Reviewer 2: ```suggestion
```

This is redundant because of line 17.

Author: Not really, it makes it easier to evaluate projects.

If they have a bazillion bugs, for example, then we may want to look at that.

Plus, for much of the software (like browsers) it has a Github button.
This makes it easier to add this if it wasn't done already.

Reviewer: Yeah, good point @ggg27, that could save a few clicks for reviewers to quickly check out the project’s source code.

Reviewer: To keep this MR moving along I'll resolve this discussion. I think this line is fine to keep in the checklist for now.
